anime_downloader/sites/darkanime.py

```diff
@@ -41,18 +41,23 @@ def getLink(self, name, _id):
 
     def _get_sources(self):
         server = self.config.get("server", "mp4upload")
-        resp = helpers.soupify(helpers.get(self.url).text).find_all('script')[-3].string
-        hosts = json.loads(re.search(r"(\[[^)]+\])", resp).group(1))
+        resp = helpers.soupify(helpers.get(self.url).text).find_all('script')#[-3].string
+        for a in resp:
+            try:

```

Comment:
Remove this try except at all costs. You said it's due to NoneType, then do something more like
```
for i in resp:
	if i:
		if 'sources' in i.string:
			res = i.string
```

Reply:
have you tried it urself? even with ur suggestion i get ERROR: argument of type 'NoneType' is not iterable
and i already tried this